resources/type-system/inference.md

```diff
@@ -1210,6 +1247,674 @@ of steps:
 _It follows, from the soundness of coercions, that the static type of `m` is
 guaranteed to be a subtype of `T`._
 
+## Argument part inference
+
+The type inference rules for many kinds of expressions make use of an operation
+known as _argument part inference_. _Argument part inference_ is a type
+inference step that is applied to an optional target function type `F`, a
+sequence of expressions `{e_1, e_2, ...}`, a corresponding sequence (of the same
+length) of optional name identifiers `{n_1, n_2, ...}`, a sequence of zero or
+more type arguments `{T_1, T_2, ...}`, and a type schema `K`, known as the
+context. Since the target function type and the names are optional, we will use
+the symbol `∅` to denote a missing target function type or a missing name. _A
+sequence of zero type arguments is typically used when supplying arguments to a
+non-generic function or method, or when type arguments need to be inferred._
+
+The output of argument part inference is a sequence of elaborated expressions
+`{m_1, m_2, ...}` (of the same length as the sequence of input expressions), a
+sequence of zero or more elaborated type arguments `{U_1, U_2, ...}`, and a
+result type known as `R`. _(The sequence of optional names is unchanged by
+argument part inference.)_
+
+To emphasize the relationship between argument part inference and the syntax of
+Dart source code, the inputs and outputs of argument part inference are
+sometimes described using the _&lt;argumentPart&gt;_ syntax, namely `<T_1, T_2,
+...>(n_1: e_1, n_2: e_2, ...)`.
+
+_So, for example, if we say that "argument part inference is invoked on `<int,
+String>(1, x: 2)`", this means that the input of argument part inference is
+applied to the sequence of expressions `{1, 2}`, the sequence of optional names
+`{∅, x}`, and the sequence of type arguments `{int, String}`._
+
+The procedure for argument part inference is as follows:
+
+- Let `{X_1, X_2, ...}` be the list of formal type parameters in `F`, or an
+  empty list if `F` is `∅`.
+
+- Let `{P_1, P_2, ...}` be the list of formal parameter types corresponding to
```

Comment:
The `P_i` must be type schemas, not types.

Reply:
Done.